src/app/core/services/basic-rundown-state.service.ts

```diff
@@ -30,75 +31,62 @@ export class BasicRundownStateService implements OnDestroy {
+  private resetBasicRundownSubjectIfConnected(isConnected: boolean): void {
     if (!isConnected) {
       return
     }
     this.resetBasicRundownSubject()
   }
 
   private resetBasicRundownSubject(): void {
-    console.log('[debug][RundownStateService]', 'Resetting basic rundowns')
     this.fetchBasicRundowns()
         .then(basicRundowns => this.basicRundownsSubject.next(basicRundowns))
         .catch(error => console.error('[error]', 'Encountered error while fetching basic rundowns:', error))
   }
 
   private registerBasicRundownEventConsumers(): Unsubscribe[] {
     return [
-      this.rundownEventObserver.subscribeToRundownActivation(this.createBasicRundownEventHandler(this.rundownActivationHandler.bind(this))),
-      this.rundownEventObserver.subscribeToRundownDeactivation(this.createBasicRundownEventHandler(this.rundownDeactivationHandler.bind(this))),
-      this.rundownEventObserver.subscribeToRundownDeletion(this.createBasicRundownEventHandler(this.rundownDeletionHandler.bind(this))),
+      this.rundownEventObserver.subscribeToRundownActivation(this.activateBasicRundownFromEvent.bind(this)),
+      this.rundownEventObserver.subscribeToRundownDeactivation(this.deactivateBasicRundownFromEvent.bind(this)),
+      this.rundownEventObserver.subscribeToRundownDeletion(this.deleteBasicRundownFromEvent.bind(this)),
       // TODO: Add listener(s) for updating the modifiedAt attribute when we implement modifiedAt.
       // TODO: Add listener(s) for when a rundown is created.
     ]
   }
 
-  private createBasicRundownEventHandler<EventType extends { rundownId: string }>(
-      handler: (basicRundown: BasicRundown, event: EventType) => BasicRundown[]
-  ): (event: EventType) => void {
-    return (event: EventType) => {
-      const basicRundowns = this.basicRundownsSubject.value
-      const newBasicRundowns: BasicRundown[] = basicRundowns.reduce(this.createBasicRundownReducer(handler, event), [])
-      this.basicRundownsSubject.next(newBasicRundowns)
-    }
-  }
-
-  private createBasicRundownReducer<EventType extends { rundownId: string }>(
-      handler: (basicRundown: BasicRundown, event: EventType) => BasicRundown[],
-      event: EventType
-  ): (basicRundowns: BasicRundown[], basicRundown: BasicRundown) => BasicRundown[] {
-    return (basicRundowns: BasicRundown[], basicRundown: BasicRundown) => {
+  private activateBasicRundownFromEvent(event: RundownEvent): void {
+    const updatedBasicRundowns = this.basicRundownsSubject.value.map(basicRundown => {
       if (basicRundown.id !== event.rundownId) {
-        return [...basicRundowns, basicRundown]
+        return basicRundown
       }
-      return [
-          ...basicRundowns,
-          ...handler(basicRundown, event)
-      ]
-    }
-  }
-
-  private rundownActivationHandler(basicRundown: BasicRundown): BasicRundown[] {
-    return [{
-      ...basicRundown,
-      isActive: true
-    }]
+      return {
+        ...basicRundown,
+        isActive: true,
+      }
```

Comment:
What's the reason for creating an entirely new object and not just?:
```
basicRundown.isActive = true
return basicRundown
```

You could then write the implementation as:
```
this.basicRundownsSubject.value.map(basicRundown => {
  if (basicRundown.id === event.rundownId) {
    basicRundown.isActive = true
  }
  return basicRundown
})
```

Sidenote: 
Instead of having to do: `const updatedBasicRundowns = this.basicRundownsSubject.value...` all the time, you could have an instance variable that is your "source of truth" that would just be `const basicRundowns: BasicRundown[]`.

Then you don't have to access the BehaviourSubject all the time and you could write your implemenation like this:
```
private activateBasicRundown(event) {
  this.basicRundowns = this.basicRundowns.map(basicRundown => {...})
  this.behaviourSubect.next(this.basicRundowns)
}
```
Then the only responsibility of the BehaviourSubject is to emit data. Not be the container for it.

Reply:
But the basicRundowns property seems like a good idea.